where-to-travel/src/main/webapp/script.js

```diff
@@ -244,11 +244,11 @@ function addPlacesFromDirection(lat, lng, place_candidates) {
     let acceptablePlaces = [];
     let i = 0;
     for (i; i < listPlaces.size(); i++) {
-      const lat = listPlaces[i % 25].geometry.location.lat();
-      const lng = listPlaces[i % 25].geometry.location.lng();
+      const lat = listPlaces[i].geometry.location.lat();
+      const lng = listPlaces[i].geometry.location.lng();
       const destination = new google.maps.LatLng(lat, lng);
       userDestinations.push(destination);
-      if (i == 24 || i == listPlaces.size() - 1) {
+      if (i == 24 || i == listPlaces.size()-1) {
```

Comment:
I believe this will still provide too many destinations to the distance matrix api if the size of listPlaces is more than 25. Maybe moving the majority of filterByDistance into a new function called addAcceptablePlaces and then doing something like 
"
 let i;
    for (i = 0; i < listPlaces.length; i += 25) {
      acceptablePlaces = await addAcceptablePlaces(time, listPlaces.slice(i, i + 25), acceptablePlaces);
    }
"

inside filterByDistance could work?

Reply:
I definitely think that could work. What about if the if loop is:  if(i%25 == 24 || i == listPlaces.size()-1)

Reply:
I think i will always equal listPlaces.size() by the time it reaches this line because that's the condition that the for loop at 257 breaks on

Reply:
Could you explain a bit more why you think it will always equal listPlaces.size() I don't quite understand.

Reply:
Oh i'm so sorry. With the way the diff looks like, I thought that the if statement happened outside of the for loop. I now see that's not the case. Then I guess my only suggestion would be to have the check be if (i % 25 == 24 || i == listPlaces.size() - 1). Sorry again!

Reply:
Ok no worries at all:)